run.sh

```diff
@@ -28,8 +28,8 @@ while [[ $# > 0 ]]; do
 done
 
 # Create shared folders
-mkdir -p $SOURCE
-mkdir -p $CCACHE
+test -d $SOURCE || mkdir -p $SOURCE
```

Comment:
I would advise against this. It makes code less legible and doesn't provide any benefit.